app/waf/src/models/action.model.ts

```diff
@@ -55,13 +66,23 @@ export class Action extends CommonEntity {
 
   @property({
     type: 'string',
-    required: true,
+    required: false,
+    schema: {
+      create: true,
+      response: true,
+      required: true,
+      example: 'httpUri',
+    },
   })
   type: string;
 
   @property({
     type: 'string',
-    required: true,
+    required: false,
```

Comment:
It should be required by repository layer, and is not required in API layer, right? We may have two styles of API to create 'has-many' resources. For example, In API layer, POST /policy/{policyId}/rules may not require policyId in body, but POST /rules does require plicyId in body. However, in db layer, plicyId must be required, since it is reference to policy.